api/users/views.py

```diff
@@ -63,7 +64,18 @@ def get_user(self, check_permissions=True):
         return obj
 
 
-class UserList(JSONAPIBaseView, generics.ListAPIView, ODMFilterMixin):
+class UserFilter(NewDjangoFilterMixin):
+
+    full_name = django_filters.CharFilter(name='fullname', lookup_expr='icontains')
+    id = django_filters.CharFilter(name='guids___id')
+
+    class Meta:
+        model = OSFUser
+        fields = ['id', 'full_name', 'given_name', 'middle_names', 'family_name']
+        strict = django_filters.constants.STRICTNESS.RAISE_VALIDATION_ERROR
```

Comment:
Would it make sense to define `strict` on the base class (currently NewDjangoFilterMixin)?